opentracing/README.md

```diff
@@ -0,0 +1,121 @@
+# DCP RFC
+
+For background:
+- [Main DCP doc](https://paper.dropbox.com/doc/Distributed-Context-Propagation-RGvlvD1NFKYmrJG9vGCES)
+- [DCP "Layers" doc](https://paper.dropbox.com/doc/DCP-Layers-and-Semantics-X1Tm1MSiBJECBkWYQKS2I)
+
+## API overview for those adding instrumentation
+
+Everyday consumers of this `opentracing` package really only need to worry
+about a couple of key abstractions: the `StartSpan` function, the `Span`
+interface, and golang's `context.Context` idiom. Here are code snippets
+demonstrating some important use cases.
+
+A note about `context.Context`: the opentracing API here encourages (but does
+not require) the use of golang-team's `context.Context` scheme. Whether we like
+it or not (this author is on the fence), `context.Context` is here to stay,
+goroutine-local-storage bedamned. When in Rome, do as the Romans do (etc).
+
+#### Singleton initialization
+
+The simplest starting point is `./global.go`. As early as possible, call
+
+    import ".../opentracing"
+    
+    func main() {
+        tracingRecorder := ... // tracing impl specific
+        tracingContextIDSource := ... // tracing impl specific
+        opentracing.InitGlobalTracer(tracingRecorder, tracingContextIDSource)
+        ...
+    }
+
+##### Note: the singletons are optional
+
+If global singletons make you sad, use `opentracing.NewStandardTracer(...)`
+directly and manage ownership of the `opentracing.OpenTracer` explicitly.
+
+#### Creating a Span given an existing Golang `context.Context`
+
+    func xyz(ctx context.Context, ...) {
+        ...
+        sp, ctx := opentracing.StartSpan("span_name", ctx)
```

Comment:
Yeah, the [pending, not-sent-out-yet] `ProcessRecorder` interface has the following method:

```
	// Every process in the system must have a name. It is up to the
	// ProcessRecorder implementation to determine how this name is specified.
	ProcessName() string
```

My concern with capturing the component is just that it may be too much to require of programmers who are in the middle or the bottom of the stack... e.g., if I am writing an ORM thats sits above a database, what component am I in? The ORM component, or some particular web service that called a model layer that then called me? There is the problem of passing the strings around, of course, but also just the need to explain this additional concept...

I'm not sure I can make a terribly strong argument against it, honestly. It just feels like we are already asking programmers to be awfully patient as we explain new terminology and concepts to them, so I want to minimize that unless there's a *clear* need for additional "required concepts"... that said, I am 100% in favor of adding a "component" tag to spans where appropriate and building UIs that make it easy to filter and aggregate across components.

Happy to talk more about this if you want, and let me know if it seems like I'm misunderstanding something.